modules/host/storagefolders.go

```diff
@@ -0,0 +1,594 @@
+	if (offloadErr != nil && offloadErr != errIncompleteOffload) || (offloadErr != nil && !force) {
+		return offloadErr
+	}
+
+	// Remove the storage folder from the host and then save the host.
+	oldStorageFolders := h.storageFolders
+	if removalIndex == len(h.storageFolders)-1 {
+		h.storageFolders = h.storageFolders[0:removalIndex]
+	} else {
+		h.storageFolders = append(h.storageFolders[0:removalIndex], h.storageFolders[removalIndex+1:]...)
+	}
+	err := h.save()
+	if err != nil {
+		// Revert the storage folders to the old storage folders, since the
+		// save operation has failed. Cap the size of the removalFolder so that
+		// future negotiations will not try to use it to store sectors.
+		//
+		// This is done because a future save may succeed, but the host needs
+		// to be kept consistent.
+		h.storageFolders = oldStorageFolders
+		removalFolder.Size = removalFolder.SizeRemaining
+		removalFolder.SizeRemaining = 0
+
+		// Check if the offload error needs to be composed into the save error.
+		if offloadErr != nil {
+			// There was an error in the offloading process, and that needs to
+			// be composed into the return value.
+			return errors.New(offloadErr.Error() + " and " + err.Error())
+		}
+		return err
+	}
+
+	// Remove the symlink that points to the data folder.
+	err = os.Remove(filepath.Join(h.persistDir, removalFolder.uidString()))
+	if err != nil && offloadErr != nil {
+		// Need to compose the offload error with the remove error.
+		return errors.New(offloadErr.Error() + " and " + err.Error())
+	} else if err != nil {
+		return err
+	}
+	return offloadErr
+}
+
+// ResizeStorageFolder changes the amount of disk space that is going to be
+// allocated to a storage folder.
+func (h *Host) ResizeStorageFolder(storageFolderIndex int, newSize uint64) error {
+	// Lock the host for the duration of the resize operation - it is important
+	// that the host not be manipulated while sectors are being moved around.
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	// The resource lock is required as the sector movements require access to
+	// the logger.
+	h.resourceLock.RLock()
+	defer h.resourceLock.RUnlock()
+	if h.closed {
+		return errHostClosed
+	}
+
+	// Check that the inputs are valid.
+	if storageFolderIndex >= len(h.storageFolders) || storageFolderIndex < 0 {
+		return errBadStorageFolderIndex
+	}
+	resizeFolder := h.storageFolders[storageFolderIndex]
+	if newSize > maximumStorageFolderSize {
+		return errLargeStorageFolder
+	}
+	if newSize < minimumStorageFolderSize {
+		return errSmallStorageFolder
+	}
+	if resizeFolder.Size == newSize {
+		return errNoResize
+	}
+
+	// Sectors do not need to be moved onto or away from the resize folder if
+	// the folder is growing, or if after being shrunk the folder still has
+	// enough storage to house all of the sectors it currently tracks.
+	resizeFolderSizeConsumed := resizeFolder.Size - resizeFolder.SizeRemaining
+	if resizeFolder.Size < newSize || resizeFolderSizeConsumed <= newSize {
```

Comment:
`resizeFolderSizeConsumed` is strictly less than `resizeFolder.Size`, so you only need to check for `resizeFolderSizeConsumed <= newSize`